Assets/Scripts/PotionSlot.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UIElements;
+
+public class PotionSlot : MonoBehaviour, IDropHandler
+{
+    [SerializeField] private PotionCollection _potionCollection;
+
+    private Potion _currentPotion;
+
+    public void OnDrop(PointerEventData eventData)
+    {
+        Potion potion = eventData.pointerDrag.GetComponent<Potion>();
```

Comment:
Later, you might wanna add some `null` check here, as someone might have dropped something that isn't a potion.